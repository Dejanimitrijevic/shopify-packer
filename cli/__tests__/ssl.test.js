/* eslint-env jest */
const ssl = require('../commands/ssl');
const mock = require('mock-fs');
const fs = require('fs');
const os = require('os');
const path = require('path');

beforeAll(() => {
  jest.spyOn(console, 'log').mockImplementation(() => {});
});

describe('Test ssl command', () => {
  beforeEach(() => {
    // Start with an empty file system
    mock({});
  });
  afterEach(() => {
    mock.restore();
  });
  test('ssl:make creates some files in .localhost_ssl', (done) => {
    ssl({make: true})
      .then(() => {
        assertSnapshot(path.join(os.homedir(), '.localhost_ssl'));
      })
      .then(done);
  });
});

function assertSnapshot(dir) {
  const result = fs.readdirSync(dir);
  mock.restore();
  expect(result).toMatchSnapshot();
}
