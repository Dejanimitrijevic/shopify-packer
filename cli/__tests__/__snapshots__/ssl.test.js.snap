// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ssl command ssl:make creates some files in .localhost_ssl 1`] = `
Array [
  "ca.crt",
  "ca.key",
  "cert.crt",
  "cert.key",
]
`;
